docs/tuf-spec.txt

```diff
@@ -928,24 +928,113 @@ Version 1.0 (Draft)
 
 5.1. The client application
 
-   1. The client application first instructs TUF to check for updates.
+    1. **Download the root metadata file**, up to some predetermined number of
+    bytes (because the size is unknown).
 
-   2. TUF downloads and verifies timestamp.json.
+    1.1. **Check signatures.** It must have been signed by a threshold of keys
+    specified in the previous root metadata file.
 
-   3. If timestamp.json indicates that snapshot.json has changed, TUF downloads
-      and verifies snapshot.json.
+    1.2. **Check for a rollback attack.** The version number of the previous
+    root metadata file must be less than or equal to the version number of this
+    root metadata file.
 
-   4. TUF determines which metadata files listed in snapshot.json differ from
-      those described in the last snapshot.json that TUF has seen. If root.json
-      has changed, the update process starts over using the new root.json.
+    1.3. **Check for a freeze attack.** The latest known time should be lower
+    than the expiration timestamp in this metadata file.
 
-   5. TUF provides the software update system with a list of available files
-      according to targets.json.
+    1.4. **If the the timestamp and / or snapshot keys have been rotated, then
+    delete the previous timestamp and snapshot metadata files.** This is done
+    in order to recover from fast-forward attacks after the repository has been
+    compromised and recovered. A _fast-forward attack_ happens when attackers
+    arbitrarily increase the version numbers of: (1) the timestamp metadata,
+    (2) the snapshot metadata, and / or (3) the targets, or a delegated
+    targets, metadata file in the snapshot metadata. Please see the Mercury
+    paper for more details (to be published soon).
```

Comment:
You can probably point to where this will be:

https://ssl.engineering.nyu.edu/papers/kuppusamy_usenix_17.pdf

For the time being, feel free to put the SUBMITTED there, but please label it DRAFT